Source/Particles/Algorithms/KineticEnergy.H

```diff
@@ -42,18 +36,14 @@ namespace Algorithms{
     {
         using namespace amrex;
 
-        constexpr auto c2 = PhysConst::c * PhysConst::c;
-        constexpr auto inv_c2 = 1.0_prt/c2;
-
-        const auto u2 = (ux*ux + uy*uy + uz*uz)*inv_c2;
-        const auto gamma = std::sqrt(1.0_prt + u2);
+        constexpr auto inv_c2 = 1.0_prt/(PhysConst::c * PhysConst::c);
 
-        const auto kk = (gamma > gamma_relativistic_threshold)?
-            (gamma-1.0_prt):
-            (u2*0.5_prt - u2*u2*(1.0_prt/8.0_prt) + u2*u2*u2*(1.0_prt/16.0_prt)-
-            u2*u2*u2*u2*(5.0_prt/128.0_prt) + (7.0_prt/256_prt)*u2*u2*u2*u2*u2); //Taylor expansion
+        // The expression used is derived by reducing the expression
+        // (gamma - 1)*(gamma + 1)/(gamma + 1)
 
-        return kk*mass*c2;
+        const auto u2 = ux*ux + uy*uy + uz*uz;
+        const auto gamma = std::sqrt(1.0_prt + u2*inv_c2);
+        return 1.0_prt/(1.0_prt + gamma)*mass*u2;
```

Comment:
I am not sure the new implementation is more precise when calculating the energy for highly non-relativistic particles.

Reply:
Hm, actually it might be a smart transition to low gamma... let's check high again